### PR TITLE
Support parsing of IPV6 addresses

### PR DESCRIPTION
**Pull request by nbagley2**

### CODE REVIEW

File: pynuodb/session.py [hunk 1]
```diff
@@ -140,8 +153,54 @@ def _split_options(options):
...
+            port = None
+            ver = ip.version
+        except ValueError:
+            # v4/v6 addr w/port e.g. 192.168.1.1:53, [2001::10]:53
+            parsed = urlparse('//{}'.format(addr))
+            try:
+                ip = self._to_ipaddr(parsed.hostname)                
+                port = parsed.port
+                ver = ip.version
+            except ValueError:
```
Comment (adriansuarez) [inline pynuodb/session.py]: It looks like this is the "low-tech" parsing attempt after `self._to_ipaddr(addr)` fails in both cases (with and without port). This only works for ipv4?

File: setup.py [hunk 2]
```diff
@@ -21,7 +21,7 @@
     url='https://github.com/nuodb/nuodb-python',
     license='BSD License',
     long_description=open(readme).read(),
-    install_requires=['pytz>=2015.4'],
+    install_requires=['pytz>=2015.4','ipaddress'],
```
[the remark this answers is not in the document]
Comment (madscientist) [reply setup.py]: I suggested this on purpose.  ipaddress is standard in Python3 but requires an additional package in Python2.  So, I just suggested that we declare it to be a requirement.  I'm not sure if there's any way in setup.py to declare different requires options for python3 vs. python2, if there is we could do that instead.

[the remark this answers is not in the document]
Comment (adriansuarez) [reply setup.py]: I guess that answers my question about `ipaddress`; any version of this package/module should work, and we just care that is exists. There does seem to be some special syntax that conditionalizes the requirement on Python version: https://stackoverflow.com/a/32643122

File: pynuodb/session.py [hunk 3]
```diff
@@ -41,6 +41,13 @@
 
 from .crypt import ClientPassword, RC4Cipher, NoCipher
 
+from ipaddress import ip_address
+
+try:
+    from urllib.parse import urlparse
```
[the remark this answers is not in the document]
Comment (nbagley2) [reply pynuodb/session.py]: python 3 has urllib.parse. python 2 has urllib

File: pynuodb/session.py (hunk 1 above)
Comment (nbagley2) [reply pynuodb/session.py]: Don't really understand question. The code as written seems to work for all the cases of v4/v6, port/no port. This was best solution I came up with so far without manually parsing every case.  The comments show which form is handled in each case. (I presume you are not looking at any of the intermediate commits; just look at the last)

Comment (adriansuarez) [reply pynuodb/session.py]: I'm referring to this particular `except`-clause. I can see first we handle "w/o port", then "with port", and then this.

Comment (nbagley2) [reply pynuodb/session.py]: The exception at line 177 is hit when we have hostname rather than literal ip addr. (i.e. test@ad0 or test@ad0:47004).

Comment (adriansuarez) [reply pynuodb/session.py]: I see. Can you add a comment to say that? Thanks.

Comment (nbagley2) [reply pynuodb/session.py]: The relevant comments are on lines 180 & 184  ('hostname w/o port...", "hostname with port...").

**Pull request merged**